Otus-DevOps-2018-02/ifqthenp_infra
# Homework 11. Ansible roles and environments (ansible-3)

# Completed Homework №11

:large_blue_diamond: [x] Main tasks
:large_orange_diamond: [x] Extra tasks for self-study

## The following tasks have been completed

- :large_blue_diamond: Created Ansible `app` and `db` roles with `ansible-galaxy init` command and assigned default variables for roles
- :large_blue_diamond: Created environments for stage and production with variables defined using Ansible `group_vars`. Added tasks for the `app` and `db` roles to display the environments they belong to
- :large_blue_diamond: Installed `jdauphant.nginx` community role with `ansible-galaxy` and configured Ansible to allow Reddit application to be reachable on port `80`. The appropriate firewall rule has been added to Terraform configuration
- :large_blue_diamond: Organised playbooks according to Ansible best practices
- :large_blue_diamond: Created `credentials.yml` for each environment and encrypted its content using `ansible-vault`
- :large_orange_diamond: Configured dynamic inventory for `stage` and `prod` environments using `gce.py`
- :large_orange_diamond: Configured `trytravis` for local repository testing in Travis CI

## How to get project running

- `cd` into `ansible` directory and run `ansible-playbook playbooks/site.yml` to start application in default stage environment

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run `google compute instances list` and enter ip-address for `reddit-app` in browser's URL bar

## PR checklisk
